java/clients/src/main/java/sleeper/clients/deploy/jar/PublishJarsToRepo.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.clients.deploy.jar;
+
+import sleeper.clients.util.command.CommandPipelineRunner;
+import sleeper.clients.util.command.CommandUtils;
+import sleeper.core.SleeperVersion;
+import sleeper.core.deploy.ClientJar;
+import sleeper.core.deploy.LambdaJar;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is used to publish built jars to a maven repository.
+ * It is called from the publishJarsToRepo.sh script in the scripts/deploy folder.
+ * Version defaults to the current SleeperVersion but can be overwritten.
+ */
+public class PublishJarsToRepo {
+    private final Path pathOfJarsDirectory;
+    private final String repoUrl;
+    private final String m2SettingsServerId;
+    private final String version;
+    private final CommandPipelineRunner commandRunner;
+    private final List<ClientJar> clientJars;
+    private final List<LambdaJar> lambdaJars;
+
+    private PublishJarsToRepo(Builder builder) {
+        this.pathOfJarsDirectory = requireNonNull(builder.pathOfJarsDirectory, "File path must not be null");
```

Comment:
This error message is still using the old name for the field, which is undescriptive.